tools/serviced-storage/driver.go

```diff
@@ -176,6 +186,118 @@ func (c *DriverList) Execute(args []string) error {
 	return nil
 }
 
+// Execute displays orphaned volumes
+func (c *CheckOrphans) Execute(args []string) error {
+	App.initializeLogging()
+	type Metadata struct {
+		CurrentDevice string
+		Snapshots     interface{}
+	}
+
+	directory := GetDefaultDriver(string(c.Args.Path))
+	volDir := directory + "/.devicemapper/volumes"
+	fsVolDir := directory + "/.devicemapper/metadata"
+
+	files, err := ioutil.ReadDir(volDir)
+	if err != nil {
+		return err
+	}
+
+	var ccVols []string
+	var snaps []string
+	var orphans []string
+	for _, f := range files {
+		if strings.Contains(f.Name(), ".") {
+			snaps = append(snaps, f.Name())
+			orphans = append(orphans, f.Name())
+		} else {
+			ccVols = append(ccVols, f.Name())
+		}
+	}
+
+	// This will store CC-visible device identifiers for comparing with the filesystem's metadata
+	var known []string
+	for _, v := range ccVols {
+		dat, _ := ioutil.ReadFile(volDir + "/" + v + "/metadata.json")
+		var m Metadata
```

Comment:
Can we pull the metadata from : https://github.com/control-center/serviced/blob/develop/volume/devicemapper/devicemapper.go#L67

Or Here: https://github.com/control-center/serviced/blob/develop/volume/devicemapper/devicemapper.go#L58